server/schedule/checker/priority_checker.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"time"
+
+	"github.com/tikv/pd/server/schedule/placement"
+
+	"github.com/tikv/pd/pkg/cache"
+	"github.com/tikv/pd/server/config"
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/schedule/opt"
+)
+
+// the default value of priority queue size
+const defaultPriorityQueueSize = 1280
+
+// PriorityChecker ensures high priority region should run first
+type PriorityChecker struct {
+	cluster opt.Cluster
+	opts    *config.PersistOptions
+	queue   *cache.PriorityQueue
+}
+
+// NewPriorityChecker creates a priority checker.
+func NewPriorityChecker(cluster opt.Cluster) *PriorityChecker {
+	return &PriorityChecker{
+		cluster: cluster,
+		opts:    cluster.GetOpts(),
+		queue:   cache.NewPriorityQueue(defaultPriorityQueueSize),
+	}
+}
+
+// GetType returns PriorityChecker's type
+func (p *PriorityChecker) GetType() string {
+	return "priority-checker"
+}
+
+// RegionPriorityEntry records region priority info
+type RegionPriorityEntry struct {
+	Retry    int
+	Last     time.Time
+	regionID uint64
+}
+
+// ID implement PriorityQueueItem interface
+func (r RegionPriorityEntry) ID() uint64 {
+	return r.regionID
+}
+
+// NewRegionEntry construct of region priority entry
+func NewRegionEntry(regionID uint64) *RegionPriorityEntry {
+	return &RegionPriorityEntry{regionID: regionID, Last: time.Now(), Retry: 1}
```

Comment:
Why does retry equal to 1 for a new entry?

Reply:
the region needs run time is : last_time+retry*10*patrol_region_interval, so the retry is 1,the region will rerun after 10*patrol_region_interval